src/DotNetLightning.Core/Channel/CommitmentsModule.fs

```diff
@@ -497,47 +497,57 @@ module RemoteForceClose =
             }
             |> Seq.sortWith TxOut.LexicographicCompare
 
-        let toRemoteIndex =
+        let toRemoteIndexOpt =
             outputs
-            |> Seq.findIndex (fun out -> out.ScriptPubKey = toRemoteScriptPubKey)
-
-        let toRemoteTxOut: TxOut = 
-            outputs 
-            |> Seq.item toRemoteIndex
-
-        let localPaymentPrivKey =
-            perCommitmentPoint.DerivePaymentPrivKey localChannelPrivKeys.PaymentBasepointSecret
-
-        (transactionBuilder.AddKeys(localPaymentPrivKey.RawKey()))
-            .AddCoins(Coin
-                          (commitments.RemoteCommit.TxId.Value,
-                           toRemoteIndex |> uint32,
-                           toRemoteTxOut.Value,
-                           toRemoteTxOut.ScriptPubKey))
-        |> ignore
+            |> Seq.tryFindIndex (fun out -> out.ScriptPubKey = toRemoteScriptPubKey)
 
-        let toLocalIndex =
+        let _spentToRemoteOutput = 
+            match toRemoteIndexOpt with
+                | None -> None
+                | Some toRemoteIndex ->
+                    let toRemoteTxOut: TxOut = 
+                        outputs 
+                        |> Seq.item toRemoteIndex
+
+                    let localPaymentPrivKey =
+                        perCommitmentPoint.DerivePaymentPrivKey localChannelPrivKeys.PaymentBasepointSecret
+
+                    (transactionBuilder.AddKeys(localPaymentPrivKey.RawKey()))
+                        .AddCoins(Coin
+                                      (commitments.RemoteCommit.TxId.Value,
+                                       toRemoteIndex |> uint32,
+                                       toRemoteTxOut.Value,
+                                       toRemoteTxOut.ScriptPubKey)) 
+                        |> ignore
+                    Some ()
+
+        let toLocalIndexOpt =
             outputs
-            |> Seq.findIndex (fun out -> out.ScriptPubKey = toLocalWitScriptPubKey)
+            |> Seq.tryFindIndex (fun out -> out.ScriptPubKey = toLocalWitScriptPubKey)
 
-        let toLocalTxOut: TxOut = 
-            outputs 
-            |> Seq.item toLocalIndex
+        let _spentToLocalOutput = 
```

Comment:
Same here about binding an unused variable. The `match` statement can just be moved to the left.